pkg/catalog/catalog.go

```diff
@@ -991,23 +990,79 @@ func (c *Catalog) DiffUncommitted(ctx context.Context, repository string, branch
 	}
 	it := NewEntryDiffIterator(iter)
 	defer it.Close()
-	return listDiffHelper(it, limit, after)
+	return listDiffHelper(it, prefix, delimiter, limit, after)
+}
+
+// GetStartPos finds the correct place to start iterating from based on delimiter, prefix & after
```

Comment:
Can you improve the documentation?  E.g. something like
```go
// GetStartPos returns the correct place to start iterating on all elements in the keys that start with `prefix' after
// `after' and taking `delimiter' into account.
```

However I still don't understand what the function does.  Specifically, for non-empty `delimiter` and `after` it does not actually return a reasonable place to start iterating, AFAICT it returns a place _after which_ to start iterating.  So perhaps
```go
// GetStartPos returns a key that SeekGE will transform to a place start iterating on all elements in the keys that start with `prefix' after
// `after' and taking `delimiter' into account
```
which is still too long.

Thing is, it's not a "correct place to start iterating" if you're going to advance past it.  So AFAIU somewhere in the godoc should say when you need to call SeekGE on the returned key.